src/main/java/org/folio/controller/AllowedServicePointsController.java

```diff
@@ -25,45 +26,61 @@
 @AllArgsConstructor
 public class AllowedServicePointsController implements AllowedServicePointsApi {
 
-  private final AllowedServicePointsService allowedServicePointsService;
+  private final ItemLevelServicePointServiceImpl itemLevelSpService;
+  private final TitleLevelServicePointServiceImpl titleLevelSpService;
 
   @Override
   public ResponseEntity<AllowedServicePointsResponse> getAllowedServicePoints(String operation,
-    UUID requesterId, UUID instanceId, UUID requestId) {
+    UUID requesterId, UUID instanceId, UUID requestId, UUID itemId) {
 
     log.info("getAllowedServicePoints:: params: operation={}, requesterId={}, instanceId={}, " +
-        "requestId={}", operation, requesterId, instanceId, requestId);
+      "requestId={}, itemId={}", operation, requesterId, instanceId, requestId, itemId);
 
     AllowedServicePointsRequest request = new AllowedServicePointsRequest(
-      operation, requesterId, instanceId, requestId);
+      operation, requesterId, instanceId, requestId, itemId);
 
     if (validateAllowedServicePointsRequest(request)) {
-      return ResponseEntity.status(OK)
-        .body(allowedServicePointsService.getAllowedServicePoints(request));
+      var response = getAllowedServicePointsResponse(request);
+      return ResponseEntity.status(OK).body(response);
     } else {
       return ResponseEntity.status(UNPROCESSABLE_ENTITY).build();
     }
   }
 
+  private AllowedServicePointsResponse getAllowedServicePointsResponse(
```

Comment:
If you are using factory method, make it return an instance of a service. Otherwise, you are calling `getAllowedServicePoints` twice here.